aiidalab_widgets_base/process.py

```diff
@@ -501,13 +500,14 @@ def update(self):
                 "Nothing to show."
             )
         else:
-            if os.path.exists(output_file_path):
-                with open(output_file_path) as fobj:
-                    difference = fobj.readlines()[
-                        len(self.output) : -1
-                    ]  # Only adding the difference
-                    self.output += difference
-                    self.value += "".join(difference)
+            if not os.path.exists(output_file_path):
```

Comment:
I added this early return to simplify the code / reduce indentation. This is an edge condition that is unlikely to happen in practice I guess.

Reply:
I think so. Why not put this path exists check to `open(output_file_path)` and put it in a try..except block?

Reply:
Yes, that would indeed be a better pattern, current code has a TOCTOU problem
https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use#Preventing_TOCTOU

But I have noticed other subtle bugs in this code (e.g. the indexing when computing the difference seems to be wrong as it would be missing the last line).
I think better course of action in the medium term is to simply copy over the corresponding widgets from QeApp, which seems to be very robust and are not well tested. I have reverted these changes to unblock this PR.

Reply:
I agree.